packages/client-core/src/util/contextParams.ts

```diff
@@ -0,0 +1,68 @@
+/*
+CPAL-1.0 License
+
+The contents of this file are subject to the Common Public Attribution License
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { LogParamsObject } from '@etherealengine/common/src/logger'
+import { EditorState } from '@etherealengine/editor/src/services/EditorServices'
+import { getState } from '@etherealengine/hyperflux'
+import { LocationState } from '../social/services/LocationService'
+import { AuthState } from '../user/services/AuthService'
+
+/**
+ * @function clientContextParams
+ * @description This function will collect contextual parameters from
+ * state's currently populated data at the moment that the function
+ * is executed.
+ *
+ * This function can't guarantee which parameters will be returned
+ * when being called, because of the nature of this helper function it
+ * can be excecuted even before any data is populated into the state's store.
+ *
+ * Along the same lines, some of these parameters will be available or not
+ * depending on the section or feature where this helper is executed.
+ *
+ * Important note: This function is expected to work as an optional hook
+ * handler for the multilogger utility in the client side, since the
+ * multilogger utility is capable of working either in the backend and in
+ * the client side, do not attempt to use this hook handler for backend logs.
+ *
+ */
+export function clientContextParams(params: LogParamsObject) {
+  const authState = getState(AuthState)
+  const locationState = getState(LocationState)
+  const editorState = getState(EditorState)
+
+  return {
+    ...params,
+    userId: authState.user?.id,
+    url: window.location.href,
+    locationId: locationState.currentLocation.location.id,
```

Comment:
i feel like getting all these states is a bit unnecessary, since all the data (aside from userId) is going to be derivable from the URL anyway

Reply:
In backend we are automatically extracting userId based on the user who is making the request from params.

https://github.com/EtherealEngine/etherealengine/blob/2b0535b1de34d7a624f323ae74e609d8a22aa9c0/packages/server-core/src/cluster/logs-api/logs-api.class.ts#L49

Reply:
@jonathan-casarrubias yeah so i think URL is sufficient